New module: xenserver_guest_facts - returns facts of XenServer VMs

bvitnik opened the pull request:
##### SUMMARY
This module implements fact gathering of XenServer VMs (guests) with module interface comparable to vmware_guest_facts module. In simple use cases, these modules are interchangeable. It complements xenserver_guest module.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
xenserver_guest_facts

##### ADDITIONAL INFORMATION
The module is trivially implemented since all the functionality is contained in xenserver module_util.

Commit:
Initial commit for xenserver_guest_facts module

 - New module: xenserver_guest_facts. Returns facts of XenServer VMs.
   Module is fully documented.

Akasurde commented:
Same like other module, could you please add unit test cases for this change - [here](https://github.com/ansible/ansible/tree/devel/test/units/modules/cloud) ? Thanks.

Commit:
Added unit tests for the module

bvitnik commented:
@Akasurde Hi. Sorry for the delay. I've added some unit tests for the module. I will need this PR merged first before I commit unit tests for other module (xenserver_guest_powerstate, PR #49425) since these two modules will share some testing code.

As this module is just using functions from the xenserver module util, I've also implemented a complete suite of unit tests for xenserver module util itself in separate PR #50850 .

Commit:
Moved FakeXenAPI import to a dedicated fixture, other fixes

bvitnik commented:
@Akasurde Hi. PR #50850 has been reviewed and merged by mattclay and I've also made appropriate changes to unit testing code in this PR. Can you please review this PR again and notify me if other changes are needed before the merge?

Akasurde commented on lib/ansible/modules/cloud/xenserver/xenserver_guest_facts.py:
```diff
@@ -0,0 +1,222 @@
+    }
+'''
+
+import re
```
Unused import.

Akasurde commented on lib/ansible/modules/cloud/xenserver/xenserver_guest_facts.py:
```diff
@@ -0,0 +1,222 @@
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text, to_native
```
Unused import

Akasurde commented on lib/ansible/modules/cloud/xenserver/xenserver_guest_facts.py:
```diff
@@ -0,0 +1,222 @@
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text, to_native
+from ansible.module_utils import six
```
Unused import.

Akasurde reviewed (changes requested):
Rest LGTM

Commit:
Removed unused imports, minor fixes to unit test code

bvitnik commented:
@Akasurde Done. Thanks for noticing unused imports.

Akasurde reviewed (approved):
LGTM
